base/algorithms.jl

```diff
@@ -0,0 +1,119 @@
+module Algorithms
```

Comment:
Seems they'd simply belong in the files corresponding to the types they're implemented for? That is, base/abstractarray.jl, base/set.jl, and base/associative.jl. I agree it doesn't make sense to separate these into their own module.